src/DataGrid.tsx

```diff
@@ -670,6 +673,7 @@ function DataGrid<R, K extends keyof R, SR>({
           rowClass={rowClass}
           top={rowIdx * rowHeight + totalHeaderHeight}
           setDraggedOverRowIdx={isDragging ? setDraggedOverRowIdx : undefined}
+          dragHandle={selectedPosition.rowIdx === rowIdx ? getDragHandle() : undefined}
```

Comment:
This looks weird combining this condition with the checks inside the `getDragHandle` function at ln#635. I think it will be better for maintenance to either add a checking outside like this

```js
dragHandle={displayDragHandle() ? getDragHandle() : undefined} // remove the checks from getDragHandle
```

or completely move all conditions inside `getDragHandle` like this

```js
function getDragHandle() {
    if (selectedPosition.rowIdx !== rowIdx || !enableCellDragAndDrop || !isCellEditable(selectedPosition) || selectedPosition.mode === 'EDIT') return null;
    ...
```

Reply:
I think it is fine. We need to add dragHandle only on the row that has the selected position. I will have to change the signature otherwise `getDragHandle(rowIdx)`. I am happy to revisit after code cleanup

Reply:
I agree with Qi. We can also do `return` instead of `return null` in `getDragHandle`.